### PR TITLE
Legend & Layers cleanup

### LINKED ISSUE #1597: [FEATURE] Legend and Layers Clean up
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Who request this Feature

_No response_

### User Story

- [ ] There too many tabbing on layer item when reorder. There shoudl be only one tabbalbe element.
- [ ] In Legend make sure all text is using translation.
- [ ] In layers, make sure all text is using translation.
- [ ] When items are unselected in layers; make sure they are hidden in legend.
- [ ] Add descriptions on the left panel for removing layer functionality.
- [ ] BUG: Points has two layers; but they are not visible on the right.
- [ ] BUG: For all layers, they should be something displayed on the right panel (unless if there is nothing to display - show a message).
- [ ] Make the attribution text look a little nice.
- [ ] Try both tabs using a different theme and make sure it they follow the theme colors.

- [ ] Make sure the points group shows child groups.

### Implementation UI

_No response_

### PR DESCRIPTION
**Pull request by cphelefu**
# Description

In this PR we are doing Legend & Layers clean-up.

We accomplish the following;
* Make sure all texts is not hardcoded (translated) in both tabs.
* Make sure Vector layers are displaying details in the layers tab.
* When the first layer is a group - it is open by default and the first valid child is selected.
* Fixing isAllItems selected in the right side of layers component.
* Replacing dummy text with valid text for remove layer description.
* Make attribution text look nice.

Fixes #1597 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.

https://cphelefu.github.io/geoview/

# Checklist:

- [x] I have build __(rush build)__ and deploy __(rush host)__ my PR
- [x] I have connected the issues(s) to this PR
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have created new issue(s) related to the outcome of this PR is needed
-  ~~I have made corresponding changes to the documentation~~
-  ~~I have added tests that prove my fix is effective or that my feature works~~
-  ~~New and existing unit tests pass locally with my changes~~

- - -
This change is [Reviewable](https://reviewable.io/reviews/Canadian-Geospatial-Platform/geoview/1605)

### CODE REVIEW

File: packages/geoview-core/src/api/event-processors/event-processor-children/legend-event-processor.ts [hunk 1]
```diff
@@ -40,10 +40,22 @@ export class LegendEventProcessor extends AbstractEventProcessor {
...
+            // TODO Adding icons list, to be verified by backend devs
+            const legendLayerListItem: TypeLegendItem = {
+              geometryType,
+              icon: iconDetailsEntry.iconImage,
+              name: iconDetailsEntry.name,
+              isVisible: 'yes',
+              default: true,
+            };
+            iconDetailsEntry.iconList = [legendLayerListItem];
+            iconDetails.push(iconDetailsEntry);
```
Comment (cphelefu) [inline packages/geoview-core/src/api/event-processors/event-processor-children/legend-event-processor.ts:58]: The change above is for making sure all layers have details. We were not pushing items for some of the layers. @ychoquet - may want to look at this code to see it makes sense.

Comment (cphelefu) [inline packages/geoview-core/src/api/event-processors/event-processor-children/legend-event-processor.ts:58]: @ychoquet @jolevesq the changes above fixes layers not displaying details. Might want to look it over.

**Review by DamonU2 — CHANGES_REQUESTED** [review]
> Reviewed 8 of 11 files at r1, 3 of 3 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/Canadian-Geospatial-Platform/geoview/1605)* status: all files reviewed, 3 unresolved discussions (waiting on @cphelefu and @jolevesq)

___
*[`packages/geoview-core/public/locales/en/translation.json` line 75 at r2](https://reviewable.io/reviews/Canadian-Geospatial-Platform/geoview/1605#-Nl_FS9p7xp0m2wd8aA5:-Nl_FS9p7xp0m2wd8aA6:b95hdtw) ([raw file](https://github.com/Canadian-Geospatial-Platform/geoview/blob/20e2429c32e43646cccc010f374cf745323b2295/packages/geoview-core/public/locales/en/translation.json#L75)):*
> ```JSON
>     "removingLayers": "Removing Layers",
>     "reArrangeLayers": "Re-arrange Layers",
>     "removeLayerDescription": "To remove a layer click on the delete icon on the right icon of each layer.",
> ```

We should probably add something to this and sorting telling how to get back to normal functionality. Also maybe change to: "To remove a layer, click on the delete icon to the right of the layer."
___
*[`packages/geoview-core/src/core/stores/store-interface-and-intial-values/layer-state.ts` line 157 at r2](https://reviewable.io/reviews/Canadian-Geospatial-Platform/geoview/1605#-Nl_IaMKCu8TdOpIT6AF:-Nl_IaMKCu8TdOpIT6AG:b-i72hpn) ([raw file](https://github.com/Canadian-Geospatial-Platform/geoview/blob/20e2429c32e43646cccc010f374cf745323b2295/packages/geoview-core/src/core/stores/store-interface-and-intial-values/layer-state.ts#L157)):*
> ```XML
>               else if (registeredLayer.style![geometryType]?.styleType === 'uniqueValue') {
>                 (registeredLayer.style![geometryType]! as TypeUniqueValueStyleConfig).uniqueValueStyleInfo[index].visible = item.isVisible;
>               }
> ```

Do we need something here and below for 'simple' styleType?

**Review by jolevesq — APPROVED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/Canadian-Geospatial-Platform/geoview/1605)* status: all files reviewed, 3 unresolved discussions (waiting on @DamonU2)

___
*[`packages/geoview-core/public/locales/en/translation.json` line 75 at r2](https://reviewable.io/reviews/Canadian-Geospatial-Platform/geoview/1605#-Nl_FS9p7xp0m2wd8aA5:-NlcPu4vD7LcAryyCEKC:bqog7yy) ([raw file](https://github.com/Canadian-Geospatial-Platform/geoview/blob/20e2429c32e43646cccc010f374cf745323b2295/packages/geoview-core/public/locales/en/translation.json#L75)):*
<details><summary><i>Previously, DamonU2 (Damon Ulmi) wrote…</i></summary><blockquote>

We should probably add something to this and sorting telling how to get back to normal functionality. Also maybe change to: "To remove a layer, click on the delete icon to the right of the layer."
</blockquote></details>

Agree!
___
*[`packages/geoview-core/src/core/stores/store-interface-and-intial-values/layer-state.ts` line 157 at r2](https://reviewable.io/reviews/Canadian-Geospatial-Platform/geoview/1605#-Nl_IaMKCu8TdOpIT6AF:-NlcQ2mVEzo-UewL6EL6:b7a5les) ([raw file](https://github.com/Canadian-Geospatial-Platform/geoview/blob/20e2429c32e43646cccc010f374cf745323b2295/packages/geoview-core/src/core/stores/store-interface-and-intial-values/layer-state.ts#L157)):*
<details><summary><i>Previously, DamonU2 (Damon Ulmi) wrote…</i></summary><blockquote>

Do we need something here and below for 'simple' styleType?
</blockquote></details>

Yes we should with a big TODO: REFACTOR use legend event processor

**Review by cphelefu — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/Canadian-Geospatial-Platform/geoview/1605)* status: all files reviewed, 2 unresolved discussions (waiting on @DamonU2)

___
*[`packages/geoview-core/public/locales/en/translation.json` line 75 at r2](https://reviewable.io/reviews/Canadian-Geospatial-Platform/geoview/1605#-Nl_FS9p7xp0m2wd8aA5:-NldI5_78ZRvm9Gu1l2a:b-896fix) ([raw file](https://github.com/Canadian-Geospatial-Platform/geoview/blob/20e2429c32e43646cccc010f374cf745323b2295/packages/geoview-core/public/locales/en/translation.json#L75)):*
<details><summary><i>Previously, jolevesq (Johann Levesque) wrote…</i></summary><blockquote>

Agree!
</blockquote></details>

Done.
___
*[`packages/geoview-core/src/core/stores/store-interface-and-intial-values/layer-state.ts` line 157 at r2](https://reviewable.io/reviews/Canadian-Geospatial-Platform/geoview/1605#-Nl_IaMKCu8TdOpIT6AF:-NldFzplA10tu8zd3Y0F:b-896fix) ([raw file](https://github.com/Canadian-Geospatial-Platform/geoview/blob/20e2429c32e43646cccc010f374cf745323b2295/packages/geoview-core/src/core/stores/store-interface-and-intial-values/layer-state.ts#L157)):*
<details><summary><i>Previously, jolevesq (Johann Levesque) wrote…</i></summary><blockquote>

Yes we should with a big TODO: REFACTOR use legend event processor
</blockquote></details>

Done.

**Review by DamonU2 — APPROVED** [review]
> Reviewed 4 of 4 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/Canadian-Geospatial-Platform/geoview/1605)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @cphelefu)

**Review by jolevesq — APPROVED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/Canadian-Geospatial-Platform/geoview/1605)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @cphelefu)

**Review by jolevesq — APPROVED** [review]
> Reviewed 1 of 11 files at r1, 3 of 4 files at r3.
> *[Reviewable](https://reviewable.io/reviews/Canadian-Geospatial-Platform/geoview/1605)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @cphelefu)

**Review by jolevesq — APPROVED** [review]
> Reviewed 5 of 11 files at r1, 1 of 3 files at r2, 1 of 4 files at r3.
> *[Reviewable](https://reviewable.io/reviews/Canadian-Geospatial-Platform/geoview/1605)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @cphelefu)

**Review by jolevesq — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/Canadian-Geospatial-Platform/geoview/1605)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @cphelefu)

**Pull request merged**